### DISCUSSION TITLE
Developer Guide Discussion

### DISCUSSION THREAD

**Discussion by IHateMyKite** [IHateMyKite/UnforgivingDevices#104] [General]
This thread should work as support for newly created Developer Guide. If anyone has issues, question or ideas about the guide, write it here.

**Comment by kurotatsu77** [comment]
As I said in other threads, the guide should contain recommended practices (what devices should be avoided randomly added for example, like Kimy hates DCL special keys being added by other mods etc).

Also it should contain guides to adding various things to work with UD - esp and script examples to go into doc folder, full description of script properties and how they affect UD devices, how to create custom Abadon suits, etc. 

Also a good idea to describe module structure of UD - what each module do, what useful functions it has to be called, with what parameters.

**Comment by IHateMyKite** [comment]
I was thinking for a long time about tutorial for making devices with UD. But because of how it changes with almost every update, it would be pain to keep it up to date. Will try to add this tutorial after the developer one (it makes sense, as the developer tutorial will explain how to even mod skyrim, and the next one will explain how to mod UD)

**Comment by kurotatsu77** [comment]
I believe even simple short cheat sheet with all modules and functions listed would serve as great reference material and would make creating in-depth tutorials much easier.

**Reply by IHateMyKite** [reply]
Adding description of modules should be not that hard, but adding description of all functions would almost impossible as there are too many functions. I will try to look if there is not instead some way to generate documentation from comments above functions.
